src/Tooltip.tsx

```diff
@@ -0,0 +1,111 @@
+import React, { CSSProperties, forwardRef, memo } from "react";
+import type { Equals } from "tsafe";
+import { assert } from "tsafe/assert";
+import { useAnalyticsId } from "./tools/useAnalyticsId";
+import { fr } from "./fr";
+import { cx } from "./tools/cx";
+
+export type UseTooltipParams = UseTooltipParams.Common &
+    (UseTooltipParams.WithTriggerOnClick | UseTooltipParams.WithTriggerOnHover);
+
+export namespace UseTooltipParams {
+    export type Common = {
+        id?: string;
+        className?: string;
+        style?: CSSProperties;
+    };
+
+    export type WithTriggerOnClick = {
+        trigger: "click";
+        buttonLabel?: string;
+    };
+
+    export type WithTriggerOnHover = {
+        trigger?: "hover";
+        buttonLabel?: never;
+    };
+}
+
+type UseTooltipReturn = {
+    tooltipOwnerProps: {
+        "aria-describedby": string;
+        className?: string;
+    };
+    TooltipOwnerButtonComponent?: JSX.Element;
+    TooltipComponent: ({ children }: { children: string }) => JSX.Element;
```

Comment:
```suggestion
    Component: ({ children }: { children: string }) => JSX.Element;
```
We know the context, it's called from an object that'll be named tooltip someting.